components/Elevation/tests/unit/MDCMaterialElevationRespondingTests.m

```diff
@@ -0,0 +1,237 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialElevation.h"
+
+#import <XCTest/XCTest.h>
+
+@interface MDCFakeConformingMDCElevationView : UIView <MDCElevation>
+@property(nonatomic, assign, readonly) CGFloat mdc_currentElevation;
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)(CGFloat elevation);
+@property(nonatomic, assign) CGFloat elevation;
+@end
+
+@implementation MDCFakeConformingMDCElevationView
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+@interface MDCFakeConformingMDCElevationViewController : UIViewController <MDCElevation>
+@property(nonatomic, assign, readonly) CGFloat mdc_currentElevation;
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)(CGFloat elevation);
+@property(nonatomic, assign) CGFloat elevation;
+@end
+
+@implementation MDCFakeConformingMDCElevationViewController
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+@interface MDCFakeConformingMDCElevationOverrideView : UIView <MDCElevation>
+@property(nonatomic, assign, readwrite) CGFloat mdc_overrideBaseElevation;
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)(CGFloat elevation);
+@property(nonatomic, assign, readonly) CGFloat mdc_currentElevation;
+@property(nonatomic, assign) CGFloat elevation;
+@end
+
+@implementation MDCFakeConformingMDCElevationOverrideView
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+@interface MDCFakeConformingMDCElevationOverrideViewController : UIViewController <MDCElevation>
+@property(nonatomic, assign, readwrite) CGFloat mdc_overrideBaseElevation;
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)(CGFloat elevation);
+@property(nonatomic, assign, readonly) CGFloat mdc_currentElevation;
+@property(nonatomic, assign) CGFloat elevation;
+@end
+
+@implementation MDCFakeConformingMDCElevationOverrideViewController
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+@interface MDCMaterialElevationRespondingTests : XCTestCase
+@property(nonatomic, strong, nullable) UIView *fakeView;
```

Comment:
Tip: Naming: Please leave out the "fake" parts. These aren't really fake views - they are real views that conform.  A fake would be closer to a replacement for an existing *functioning* object where the fake's behavior is not complete compared to the "real" object.

Reply:
Done